bench/hist/Main.hs

```diff
@@ -259,7 +254,31 @@ main = shakeArgs shakeOptions {shakeChange = ChangeModtimeAndDigest} $ do
         title = show (unescapeExperiment exp) <> " - live bytes over time"
     plotDiagram False diagram out
 
-----------------------------------------------------------------------------------------------------
+--------------------------------------------------------------------------------
+
+buildGhcide :: BuildSystem -> String -> String
+buildGhcide Cabal out = unwords
+    ["cabal install"
+    ,"exe:ghcide"
+    ,"--installdir=" ++ out
+    ,"--install-method=copy"
+    ,"--overwrite-policy=always"
+    ,"--ghc-options -rtsopts"
+    ]
+buildGhcide Stack out =
+    "stack --local-bin-path=" <> out
+        <> " build ghcide:ghcide --copy-bins --ghc-options -rtsopts"
+
+
+findGhc :: BuildSystem -> Action FilePath
+findGhc Cabal = do
+    Stdout ghcLoc <- cmd (s "which ghc")
+    return ghcLoc
+findGhc Stack = do
+    Stdout ghcLoc <- cmd (s "stack exec which ghc")
```

Comment:
This one seems fine on the basis that stack may install which, and runs exec on its path.